model/common/src/icon4py/model/common/states/factory.py

```diff
@@ -611,13 +623,18 @@ def _compute(
         args.update(offsets)
         args.update(self._params)
         results = self._func(**args)
-        # TODO(): can the order of return values be checked?
-        results = (results,) if isinstance(results, data_alloc.NDArray) else results
+        # convert to tuple
+        results = (results,) if not isinstance(results, tuple) else results
         self._fields = {
-            k: gtx.as_field(tuple(self._dims), results[i], allocator=backend)
+            k: self._as_field(backend, results[i]) if self._dims else results[i]
```

Comment:
Just for my understanding: Is the idea even that a scalar value returned by `_func` will be converted to a "scalar"/0-d gt4py field? I.e. it does not need to be treated differently because gt4py supports scalar fields?